src/components/update/updateArticle.js

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import ReactQuill from 'react-quill';
+import '../articles/createArticles/createArticles.scss';
+import 'react-quill/dist/quill.snow.css';
+import { ToastContainer } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import PropTypes from 'prop-types';
+import { modules, formats } from '../articles/createArticles/editor';
+
+import NavBar from '../common/navbar/navbar';
+
+import {
+  flashToast,
+  getArticle,
+  updateArticle,
+} from '../../redux/actions/actionHelpers';
+
+import './update.scss';
+
+export class UpdateArticles extends React.Component {
+  constructor(props) {
+    super(props);
+    this.modules = modules;
+    this.formats = formats;
+    this.getArticle = getArticle;
+    this.state = {
+      slug: '',
+      title: '',
+      body: '',
+      description: '',
+      schem: '',
+      images: [],
+      imageWidget: cloudinary.createUploadWidget(
+        {
+          cloudName: 'nshuti-jonathan',
+          uploadPreset: 'f31rkki4',
+          multiple: false,
+          cropping: true,
+          croppingShowBackButton: true,
+          singleUploadAutoClose: false,
+        },
+        (error, result) => {
+          if (!error && result && result.event === 'success') {
+            const { images } = this.state;
+            images[0] = result.info.secure_url;
+            images[0] = images[0].toString();
+            this.setState({ images });
+          }
+        },
+      ),
+    };
+    this.handleChangeField = this.handleChangeField.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount = async () => {
+    const url = this.props.location.pathname;
+    const slug = url.split('/')[2];
+    this.setState({ slug });
+    const article = await this.props.getArticle(slug);
+    this.setState(article.data);
+    flashToast(article);
+  };
+
+  handleChange = (html) => {
+    this.setState({ body: html });
+  };
+
+  showWidget = () => {
+    const { imageWidget } = this.state;
+    imageWidget.open();
+  };
+
+  handleSubmit = async () => {
+    const { slug } = this.state;
+    const update = await this.props.updateArticle(slug, this.state);
+    this.setState(update.updatedArticle);
+    flashToast(update);
+  };
+
+  handleChangeField(event) {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  }
+
+  render() {
+    const { title, body, description } = this.state;
+    return (
+      <div className="container">
+        <NavBar />
```

Comment:
what do you think having navbar in app.js instead of calling it in every component

Reply:
Rebased and now navbar is in app. Have removed it locally, waiting for more feedback so i push changes at once